frontend/src/components/tabs/MetricsTab.tsx

```diff
@@ -0,0 +1,194 @@
+          />
+        )}
+        {errorArtifactTypes && (
+          <Banner
+            message='Error in retriving artifact types information.'
+            mode='error'
+            additionalInfo={errorArtifactTypes.message}
+          />
+        )}
+        {!isLoading && !errorArtifacts && !errorArtifactTypes && artifacts && artifactTypes && (
+          <MetricsSwitcher artifacts={artifacts} artifactTypes={artifactTypes}></MetricsSwitcher>
+        )}
+      </div>
+    </div>
+  );
+}
+
+/**
+ * Visualize system metrics based on artifact input.
+ */
+function MetricsSwitcher({ artifacts, artifactTypes }: MetricsSwitcherProps) {
+  // system.ClassificationMetrics contains confusionMatrix or confidenceMetrics.
+  // TODO: Visualize confusionMatrix using system.ClassificationMetrics artifacts.
+  // https://github.com/kubeflow/pipelines/issues/5668
+  let classificationMetricsArtifacts = filterArtifactsByType(
+    'system.ClassificationMetrics',
+    artifactTypes,
+    artifacts,
+  );
+
+  const confidenceMetricsArtifacts = classificationMetricsArtifacts
+    .map(artifact => ({
+      name: artifact
+        .getCustomPropertiesMap()
+        .get('name')
+        ?.getStringValue(),
+      customProperties: artifact.getCustomPropertiesMap(),
+    }))
+    .filter(x => !!x.name)
+    .filter(x => {
+      const confidenceMetrics = x.customProperties
+        .get('confidenceMetrics')
+        ?.getStructValue()
+        ?.toJavaScript();
+      return !!confidenceMetrics;
+    });
+
+  return (
+    <>
+      {(!confidenceMetricsArtifacts || confidenceMetricsArtifacts.length === 0) && (
+        <Banner message='There is no metrics artifact available in this step.' mode='info' />
+      )}
+      {classificationMetricsArtifacts &&
+        classificationMetricsArtifacts.length > 0 &&
+        classificationMetricsArtifacts.map(artifact => {
+          const id = artifact.getId();
+          const customProperties = artifact.getCustomPropertiesMap();
+          const confidenceMetrics = customProperties
+            .get('confidenceMetrics')
+            ?.getStructValue()
+            ?.toJavaScript();
+          return (
+            <React.Fragment key={id}>
+              {confidenceMetrics && (
+                <div>
+                  <div className={padding(40, 'lr')}>
+                    <h1>
+                      ROC Curve{' '}
+                      <IconWithTooltip
+                        Icon={HelpIcon}
+                        iconColor={color.weak}
+                        tooltip={ROC_CURVE_DEFINITION}
+                      ></IconWithTooltip>
+                    </h1>
+                  </div>
+                  <ROCCurve configs={buildRocCurveConfig((confidenceMetrics as any).list)} />
+                </div>
+              )}
+            </React.Fragment>
+          );
+        })}
+    </>
+  );
+}
+
+function buildRocCurveConfig(confidenceMetricsArray: ConfidenceMetrics[]): ROCCurveConfig[] {
```

Comment:
nit: no need to repeat type in var name

just `confidenceMetrics`

Reply:
Error as below will appear if I not specify type:

```
Parameter 'confidenceMetricsArray' implicitly has an 'any' type.ts(7006)
```

If I force remove the type specification, it will ask me to add `any` type to it. Am I understanding the suggested change mistakenly?